lib/fs.js

```diff
@@ -2379,8 +2379,13 @@ WriteStream.prototype.open = function() {
 
 
 WriteStream.prototype._write = function(data, encoding, cb) {
-  if (!(data instanceof Buffer))
-    return this.emit('error', new Error('Invalid data'));
+  if (!(data instanceof Buffer)) {
+    const err = new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                     'data',
+                                     ['Buffer'],
```

Comment:
Here can be reduced to use a single string `'Buffer'`, instead of an array `['Buffer']` : )